automotive/automotive-lib/src/main/java/com/example/android/uamp/automotive/lib/MultiSelectListPreference.kt

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.uamp.automotive.lib
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.os.Parcel
+import android.os.Parcelable
+import android.util.AttributeSet
+import androidx.annotation.ArrayRes
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.res.TypedArrayUtils
+import androidx.preference.Preference
+import java.util.Collections
+import kotlin.collections.HashSet
+
+/**
+ * A {@link Preference} that displays a list of entries in a {@link ListPreferenceFragment}.
+ *
+ * This class is taken largely as is from {@link androidx.preference.ListPreference} other than
+ * the modification to display entries in a full-screen fragment rather than a dialog.
+ *
+ * <p>This preference saves a string value. This string will be the value from the
+ * {@link #setEntryValues(CharSequence[])} array.
+ *
+ * @attr name android:entries
+ * @attr name android:entryValues
+ */
+class MultiSelectListPreference : Preference {
+
+  companion object {
+    internal const val ARG_KEY = "AutomotiveMultiSelectListPreferenceKey"
```

Comment:
UAMP tends to use top level constants rather than constants in the `companion object`.

Reply:
Done